job_board/portals/himalayas.py

```diff
@@ -50,47 +32,42 @@ def get_jobs(self) -> list[Job]:
 
         jobs_data = []
         jobs_fetched = 0
-        with httpx_client() as client:
-            while True:
-                for attempt in Retrying(
-                    stop=stop_after_attempt(5),
-                    retry=retry_if_exception(is_retryable),
-                    wait=wait_exponential(min=1, multiplier=1, max=5),
-                    reraise=True,
-                    before_sleep=before_sleep_logging,
-                ):
-                    with attempt:
-                        response = client.get(
-                            self.url,
-                            params={"offset": jobs_fetched, "limit": 20},
-                        ).json()
-
-                total_jobs = response["totalCount"]
-                job_data = response["jobs"]
-                jobs_data.extend(job_data)
-                jobs_fetched += len(job_data)
-                logger.info(
-                    (
-                        f"[Himalayas]: Fetched {jobs_fetched} of {total_jobs} jobs, "
-                        f"Remaining: {total_jobs - jobs_fetched}"
-                    )
+        while True:
+            response = self.make_request(params={"offset": jobs_fetched, "limit": 20})
+            total_jobs = response["totalCount"]
+            job_data = response["jobs"]
+            jobs_data.extend(job_data)
+            jobs_fetched += len(job_data)
+            logger.info(
+                (
+                    f"[Himalayas]: Fetched {jobs_fetched} of {total_jobs} jobs, "
+                    f"Remaining: {total_jobs - jobs_fetched}"
                 )
+            )
 
-                # no need to make any more requests if we have
-                # already fetched all the jobs that were posted
-                # after the last run or all jobs are too old.
-                if all(cutoff_date > self.get_posted_on(j) for j in job_data):
-                    logger.info(
-                        f"[Himalayas]: No more jobs to fetch. "
-                        f"{cutoff_date=}, {jobs_fetched=}"
-                    )
-                    break
+            # no need to make any more requests if we have
+            # already fetched all the jobs that were posted
+            # after the last run or all jobs are too old.
+            if all(cutoff_date > self.get_posted_on(j) for j in job_data):
+                logger.info(
+                    f"[Himalayas]: No more jobs to fetch. "
+                    f"{cutoff_date=}, {jobs_fetched=}"
+                )
+                break
 
-                if jobs_fetched >= total_jobs:
-                    break
+            if jobs_fetched >= total_jobs:
+                break
 
         return self.filter_jobs(jobs_data)
 
+    @retry_on_http_errors()
+    def make_request(self, params):
```

Comment:
I would have ideally wanted to use this a context manager, instead of having to move the logic to make request to a separate method. 

But couldn't get it working without too much complexity.